src/v/pandaproxy/schema_registry/types.h

```diff
@@ -116,4 +116,59 @@ struct subject_schema {
     bool deleted{false};
 };
 
+enum class compatibility_level {
+    none = 0,
+    backward,
+    backward_transitive,
+    forward,
+    forward_transitive,
+    full,
+    full_transitive,
+};
+
+constexpr std::string_view to_string_view(compatibility_level v) {
+    switch (v) {
+    case compatibility_level::none:
+        return "NONE";
+    case compatibility_level::backward:
+        return "BACKWARD";
+    case compatibility_level::backward_transitive:
+        return "BACKWARD_TRANSITIVE";
+    case compatibility_level::forward:
+        return "FORWARD";
+    case compatibility_level::forward_transitive:
+        return "FORWARD_TRANSITIVE";
+    case compatibility_level::full:
+        return "FULL";
+    case compatibility_level::full_transitive:
+        return "FULL_TRANSITIVE";
+    }
+    return "{invalid}";
```

Comment:
I'm not sure I understand. The `unreachable` call makes gcc happy, and we get errors if all the cases aren't covered.

Reply:
> I'm not sure I understand. The `unreachable` call makes gcc happy, and we get errors if all the cases aren't covered.

What happens if it's ever hit, though? Doesn't gcc assume UB, and make potentially wild optimisations based on it? I guess it could only do that if it could const fold through everything? If there's no match, and it hits `__builtin_unreachable()` at runtime, does it corrupt the stack?

Reply:
It results in a segfault.

Reply:
I'll handle it later, there are a few of these: https://github.com/vectorizedio/redpanda/issues/1720